### PR TITLE
compiler: use "github.com/dlclark/regexp2" instead of stdlib regexp

### PR DESCRIPTION
**Pull request by ayzu**
Package "github.com/dlclark/regexp2" supports negative lookahead,
which is required by https://github.com/ory/oathkeeper/issues/321

Signed-off-by: Aynur Zulkarnaev <ayzulkarnaev@gmail.com>

### CODE REVIEW

**Comment by ayzu** [comment]
Cool, thank you. I will ping you as I finish.

**Comment by ayzu** [comment]
@aeneasr Hi, I finished.

**Comment by aeneasr** [comment]
Thank you @ayzu ! I will review it by tomorrow!

File: compiler/regex.go [hunk 1]
```diff
@@ -125,7 +127,7 @@ func CompileRegex(tpl string, delimiterStart, delimiterEnd byte) (*regexp.Regexp
 		// Build the regexp pattern.
 		varIdx := i / 2
 		fmt.Fprintf(pattern, "%s(%s)", regexp.QuoteMeta(raw), patt)
-		varsR[varIdx], err = regexp.Compile(fmt.Sprintf("^%s$", patt))
+		varsR[varIdx], err = regexp2.Compile(fmt.Sprintf("^%s$", patt), regexp2.None)
```
Comment (aeneasr) [inline compiler/regex.go]: Why did you choose `None` here? Wouldn't that make it backwards incompatible?

File: compiler/regex_test.go [hunk 2]
```diff
@@ -45,8 +45,11 @@ func TestRegexCompiler(t *testing.T) {
 		{"urn:foo.bar.com:{.*{}", '{', '}', true, "", true},
 		{"urn:foo:<.*>", '<', '>', false, "urn:foo:bar:baz", false},
 
+		{`urn:foo:<user=(?!admin).*>`, '<', '>', false, "urn:foo:user=john", false},
+		{`urn:foo:<user=(?!admin).*>`, '<', '>', false, "urn:foo:user=admin", true},
+
```
Comment (aeneasr) [inline compiler/regex_test.go]: Could you please add a test for backwards compatible regex here? You could pick one of "named ascii character class [[:foo:]]" or "Python-style capture groups (P<name>re)" or both.

**Comment by ayzu** [comment]
Yes, sure. Thank you for the review, I will update the code.

File: compiler/regex_test.go (hunk 2 above)
Comment (ayzu) [reply compiler/regex_test.go]: OK, added.

File: compiler/regex.go (hunk 1 above)
Comment (ayzu) [reply compiler/regex.go]: Yes, thank you. Switched to another option.

**Comment by ayzu** [comment]
@aeneasr I have updated the code, please have a look.

**Comment by aeneasr** [comment]
I have released these changes as `v1.1.0`! You can now import them in oathkeeper!

**Pull request merged**